synapse/replication/http/_base.py

```diff
@@ -201,6 +214,10 @@ def make_client(cls, hs: "HomeServer") -> Callable:
 
         @trace_with_opname("outgoing_replication_request")
         async def send_request(*, instance_name: str = "master", **kwargs: Any) -> Any:
+            # We have to pull these out here to avoid circular dependencies...
+            streams = hs.get_replication_command_handler().get_streams_to_replicate()
+            replication = hs.get_replication_data_handler()
```

Comment:
To check: we can't use `self._streams` and `self._replication` because they'll only exist if we're receiving replication data?

Reply:
Yup. This is a class method, essentially, so we don't have access to `self`.